calculation/spatial_hazard/spatial_hazard/correlate_ims.py

```diff
@@ -0,0 +1,227 @@
+from typing import Sequence
+
+import numpy as np
+import pandas as pd
+from scipy.linalg import cholesky
+
+import gmhazard_calc as gc
+from spatial_hazard import models
+from IM_calculation.source_site_dist.src_site_dist import calc_rrup_rjb
+
+
+def load_stations_data(imdb_ffps: Sequence[str], stations: Sequence[str], im: gc.im.IM):
+    """Loads the IM data for the specified stations from the IMDBs"""
+    result_dict = {station: [] for station in stations}
+    for imdb_ffp in imdb_ffps:
+        print("Getting {}".format(imdb_ffp))
+        with gc.dbs.IMDB.get_imdb(imdb_ffp, writeable=False) as imdb:
+            imdb_stations = imdb.sites().index
+            for station in stations:
+                if station not in imdb_stations:
+                    print("Station {} has no entries here".format(station))
+                cur_data = imdb.im_data(station, im)
+                if cur_data is not None:
+                    result_dict[station].append(cur_data)
+
+    result_dict = {key: pd.concat(value) for key, value in result_dict.items()}
+    return result_dict
+
+
+def load_stations_fault_data(
+    imdb_ffps: Sequence[str], stations: Sequence[str], im: gc.im.IM, fault: str
+):
+    """Loads the IM data for the specified stations and fault from the IMDBs"""
+    stations_data = load_stations_data(imdb_ffps, stations, im)
+
+    return pd.DataFrame(
+        {
+            cur_station: cur_df.loc[fault].to_dict()
+            for cur_station, cur_df in stations_data.items()
+        }
+    ).T
+
+
+def calculate_distance_matrix(
+    stations: Sequence[str], locations_df: pd.DataFrame
+) -> pd.DataFrame:
+    """
+    Given a set of stations and their locations (in lat, lon format), calculate the matrix containing
+    the pairwise distance between each one
+
+    Parameters
+    ----------
+    stations: List[str]
+        List of the stations
+    locations_df: pd.DataFrame
+        Locations of each of the stations (in lat, lon)
+
+    Returns
```

Comment:
If we decide to have Returns, do we also need to specify what we return?
I never used Returns in the docstring, so not sure what to put haha
@claudio525 @jasonmotha